torch/csrc/jit/export.cpp

```diff
@@ -16,222 +17,177 @@ namespace torch { namespace jit {
-      auto t = attr->mutable_t();
-      encodeTensor(t, n->t(name));
-    } break;
-    case AttributeKind::ts:
-      attr->set_type(onnx::aTENSORS);
-      for(auto & v : n->ts(name)) {
-        auto t = attr->add_tensors();
-        encodeTensor(t, v);
-      }
-      break;
-    case AttributeKind::g: {
-      attr->set_type(onnx::aGRAPH);
-      auto g = attr->mutable_g();
-      encodeGraph(g, n->g(name), {}, ctx, nullptr);
-    } break;
-    case AttributeKind::gs:
-      attr->set_type(onnx::aGRAPHS);
-      for(auto & v : n->gs(name)) {
-        auto g = attr->add_graphs();
-        encodeGraph(g, v, {}, ctx, nullptr);
-      }
-      break;
-  }
-}
-
-void encodeTypeProtoTensorType(onnx::TypeProtoTensor* tensor_type, Value* n) {
+void EncodeTypeProtoTensorType(onnx::TypeProto_Tensor* tensor_type, const Value* n) {
   onnx::TensorShapeProto* shape = tensor_type->mutable_shape();
   if (TensorType* node_type = n->type()->cast<TensorType>()) {
     const std::vector<std::int64_t>& sizes = node_type->sizes();
-    for (std::int64_t s : sizes) {
-      shape->add_dim(s);
-    }
-    onnx::DataType onnx_type;
-    switch(node_type->scalarType()) {
-      case at::kDouble:
-        onnx_type = onnx::kDOUBLE;
-        break;
-      case at::kFloat:
-        onnx_type = onnx::kFLOAT;
-        break;
-      case at::kHalf:
-        onnx_type = onnx::kFLOAT16;
-        break;
-      case at::kByte:
-      case at::kChar:
-        onnx_type = onnx::kINT8;
-        break;
-      case at::kShort:
-        onnx_type = onnx::kINT16;
-        break;
-      case at::kInt:
-        onnx_type = onnx::kINT32;
-        break;
-      case at::kLong:
-        onnx_type = onnx::kINT64;
-        break;
-      default:
-        torch::barf("unexpected tensor scalar type");
-        break;
+    for (size_t i = 0; i < sizes.size(); i++) {
+      shape->add_dim();
+      shape->mutable_dim(i)->set_dim_value(sizes[i]);
     }
-    tensor_type->set_data_type(onnx_type);
+    tensor_type->set_elem_type(ATenTypeToOnnxType(node_type->scalarType()));
   }
 }
 
-void encodeValueInfo(onnx::ValueInfoProto* v, Value* n) {
-  v->set_name(value_name(n));
+void EncodeValueInfo(onnx::ValueInfoProto* v, const Value* n) {
+  v->set_name(n->uniqueName());
   onnx::TypeProto* t = v->mutable_type();
-  onnx::TypeProtoTensor* tensor_type = t->mutable_tensor_type();
-  encodeTypeProtoTensorType(tensor_type, n);
+  onnx::TypeProto_Tensor* tensor_type = t->mutable_tensor_type();
+  EncodeTypeProtoTensorType(tensor_type, n);
+}
+
+JitEncoder::JitEncoder(
+    onnx::ModelProto *model_proto,
+    int64_t onnx_opset_version,
+    onnx_torch::OperatorExportTypes operator_export_type,
+    bool defer_weight_export)
+    : num_blocks_(0),
+      defer_weight_export_(defer_weight_export),
+      operator_export_type_(operator_export_type) {
+  model_proto->set_producer_name("pytorch");
+  model_proto->set_ir_version(3);
+  model_proto->set_producer_version("0.4");
```

Comment:
maybe do set_doc_string() or change set_producer_name() to signify it's not a real onnx?